ek9000App/src/devEL1XXX.cpp

```diff
@@ -124,19 +121,13 @@ template <class RecordT> static long EL10XX_read_record(void* prec) {
 																// channel is 1-based index, m_inputStart is also
 																// 1-based, but modbus coils are 0-based, hence the -2
 	assert(num <= sizeof(buf));
-	status = dpvt->pterm->getEK9000IO(MODBUS_READ_DISCRETE_INPUTS, addr, buf, num);
+	int status = dpvt->pterm->getEK9000IO(READ_DIGITAL, addr, buf, num);
 
 	/* Error states */
-	if (status) {
-		recGblSetSevr(pRecord, READ_ALARM, INVALID_ALARM);
-		/* Check type of err */
-		if (status > 0x100) {
-			LOG_WARNING(dpvt->pdrv, "EL10XX_read_record() for %s: %s\n", pRecord->name,
-						devEK9000::ErrorToString(status));
-			return 0;
-		}
-		LOG_WARNING(dpvt->pdrv, "EL10XX_read_record() for %s: %s\n", pRecord->name, devEK9000::ErrorToString(status));
-		return 0;
+	if (status != EK_EOK) {
+		recGblSetSevr(pRecord, COMM_ALARM, INVALID_ALARM);
```

Comment:
On the client side, we care most about the alarm severity which is set appropriately here as `INVALID`. 

It's possible others may disagree, but in my opinion the alarm status of "READ" vs "COMM" is just useful metadata for the user to differentiate between different potential scenarios. "Communication with the controller failed" vs "my attempt to read data failed (for some other reason)" is how I'd interpret them.

Reply:
Thanks for the input! I was unsure of this change and was totally unaware of the other alarm types until last week when I was reading some random device support code.